lucene/core/src/java/org/apache/lucene/index/EagerCheapMergePolicy.java

```diff
@@ -0,0 +1,204 @@
+          infoStream.message(INFO_STREAM_CATEGORY,
+              "Found a cheap merge: " + toSummaryString(cheapMerge.segments));
+          //FYI further IW infoStream message "registerMerge" will have segment details
+        }
+        MergeSpecification spec = new MergeSpecification();
+        spec.add(cheapMerge);
+        return spec;
+      }
+    } else {
+      if (verbose) {
+        infoStream.message(INFO_STREAM_CATEGORY,
+            "Don't look for a cheap merge because " +
+                mergeContext.getMergingSegments().size() + " segments are already merging.");
+      }
+    }
+    return null;
+  }
+
+  private String toSummaryString(List<SegmentCommitInfo> segments) {
+    long sizeBytes = segments.stream().mapToLong(EagerCheapMergePolicy::size).sum();
+    return String.format(Locale.ROOT, "segCount=%d sizeMB=%.3f", segments.size(), sizeBytes / 1024.0 / 1024.0);
+  }
+
+  private OneMerge findCheapMerge(SegmentInfos infos, MergeContext mergeContext) throws IOException {
+    List<SegmentCommitInfo> cheapMerge = findCheapMerge(infos.asList(), mergeContext.getMergingSegments());
+    return cheapMerge == null ? null : new OneMerge(cheapMerge);
+  }
+
+  //@VisibleForTesting
+  List<SegmentCommitInfo> findCheapMerge(List<SegmentCommitInfo> segmentInfoPerCommits,
+                                         Collection<SegmentCommitInfo> alreadyMerging) {
+
+    // Find segments that are below the size threshold and that aren't being merged already
+    final List<SegmentCommitInfo> infosSortedAsc = segmentInfoPerCommits.stream()
+        .filter(o -> !alreadyMerging.contains(o))
+        .filter(o -> size(o) <= cheapMergeThresholdBytes)
+        .sorted(Comparator.comparingLong(EagerCheapMergePolicy::size))
+        .collect(Collectors.toList());
+
+    // Maximize how many of these we can merge while keeping in budget
+    long accumSize = 0L;
+    int numSegmentsToMerge = 0;
+    for (int segIdx = 0; segIdx < infosSortedAsc.size(); segIdx++) {
+      SegmentCommitInfo infoPerCommit = infosSortedAsc.get(segIdx);
+      long segSize = size(infoPerCommit);
+      if (accumSize + segSize > cheapMergeThresholdBytes) {
+        break;
+      }
+      // if the ratio of this segment's size to the sum of sizes of segments we've seen is less than
+      //   a threshold, then decide we will merge this (along with all before).
+      // note: on 1st loop, accumSize==0; division results in: Infinity < whatever (false)
+      if (((double) (segSize)) / ((double) accumSize) < cheapMaxSizeRatio) {
+        numSegmentsToMerge = segIdx + 1;
+      }
+      accumSize += segSize;
+    }
+    if (numSegmentsToMerge < cheapMinMergeAtOnce) {
+      return null; // too few segments to merge
+    }
+    return infosSortedAsc.subList(0, numSegmentsToMerge);
+  }
+
+  private static long size(SegmentCommitInfo infoPerCommit) {
+    //note: We could discount for deleted docs (as TieredMergePolicy does) but lets keep this simple/cheap.
```

Comment:
But it is important for a merge policy to favor merging segments with many deletions?  Or, are you thinking that on commit the newly flushed segments will not have many deletions typically so why bother?  I think it would be simpler / less risky to just use the existing `MergePolicy.size`?